torchrl/collectors/collectors.py

```diff
@@ -526,15 +578,25 @@ def iterator(self) -> Iterator[TensorDictBase]:
         Yields: TensorDictBase objects containing (chunks of) trajectories
 
         """
-        total_frames = self.total_frames
         i = -1
         self._frames = 0
         while True:
             i += 1
             self._iter = i
             tensordict_out = self.rollout()
-            self._frames += tensordict_out.numel()
-            if self._frames >= total_frames:
+
+            # Bring all batch dimensions to the front (only performs computation if it is not already the case)
+            tensordict_out = tensordict_out.permute(self.permute_out_batch_size)
+            # Flatten all batch dimensions into first one and leave unmasked dimensions untouched
+            if len(self.env_batch_size_unmasked) > 0:
+                tensordict_out = tensordict_out.reshape(
+                    self.frames_per_batch * self.n_env, *self.env_batch_size_unmasked
+                )
+            else:
+                tensordict_out = tensordict_out.view(-1).to_tensordict()
```

Comment:
I order to move all batch dimensions to the first, I use first a permute and then a reshape. 
This makes it impossible to `return_same_td`.
Is there a workaround to this returning the same td? Or what can we do?